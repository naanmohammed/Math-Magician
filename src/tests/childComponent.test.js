import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import ChildComponent from '../components/childComponent';

describe('ChildComponent', () => {
  const mockHandleButtonClick = jest.fn();
  const mockCalcData = {
    total: '10',
    next: null,
    operation: '+',
  };

  it('renders the component', () => {
    render(
      <ChildComponent calcData={mockCalcData} handleButtonClick={mockHandleButtonClick} />
    );

    const calculatorElement = screen.getByRole('textbox');
    expect(calculatorElement).toBeInTheDocument();
  });

  it('calls handleButtonClick when a button is clicked', () => {
    const handleButtonClickMock = jest.fn();
    render(<ChildComponent calcData={{}} handleButtonClick={handleButtonClickMock} />);
    const buttonElement = screen.getByRole('button', { name: '7' });
    fireEvent.click(buttonElement);
    expect(handleButtonClickMock).toHaveBeenCalledWith('7');
  });
});
