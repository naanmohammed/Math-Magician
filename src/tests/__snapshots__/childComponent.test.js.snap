// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildComponent renders the component 1`] = `
<div
  class="calculator-container"
>
  <input
    class="calculator-input-field"
    id="myInput"
    name="name"
    pattern="[0-9+\\\\-*/(). ]+"
    type="text"
    value="+"
  />
  <div
    class="calculator-buttons"
  >
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  "
        id="AC"
        type="button"
      >
        AC
      </button>
      <button
        class="calculator-button  "
        id="+/-"
        type="button"
      >
        +/-
      </button>
      <button
        class="calculator-button  "
        id="%"
        type="button"
      >
        %
      </button>
      <button
        class="calculator-button math-button-orange "
        id="รท"
        type="button"
      >
        รท
      </button>
    </div>
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  "
        id="7"
        type="button"
      >
        7
      </button>
      <button
        class="calculator-button  "
        id="8"
        type="button"
      >
        8
      </button>
      <button
        class="calculator-button  "
        id="9"
        type="button"
      >
        9
      </button>
      <button
        class="calculator-button math-button-orange "
        id="x"
        type="button"
      >
        x
      </button>
    </div>
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  "
        id="4"
        type="button"
      >
        4
      </button>
      <button
        class="calculator-button  "
        id="5"
        type="button"
      >
        5
      </button>
      <button
        class="calculator-button  "
        id="6"
        type="button"
      >
        6
      </button>
      <button
        class="calculator-button math-button-orange "
        id="-"
        type="button"
      >
        -
      </button>
    </div>
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  "
        id="1"
        type="button"
      >
        1
      </button>
      <button
        class="calculator-button  "
        id="2"
        type="button"
      >
        2
      </button>
      <button
        class="calculator-button  "
        id="3"
        type="button"
      >
        3
      </button>
      <button
        class="calculator-button math-button-orange "
        id="+"
        type="button"
      >
        +
      </button>
    </div>
    <div
      class="calculator-row"
    >
      <button
        class="calculator-button  zero-button"
        id="0"
        type="button"
      >
        0
      </button>
      <button
        class="calculator-button  "
        id="."
        type="button"
      >
        .
      </button>
      <button
        class="calculator-button  "
        id="="
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
